### PR TITLE
Bring extension up to date with MW standards

### PR DESCRIPTION
**Pull request by reedy**

### CODE REVIEW

**Comment by ernstki** [comment]
@reedy I'm happy to merge this in, but I'm not clear on what was changed and why.

I've never contributed to MediaWiki before, and someone _else_ modified [Extension:PlaceNewSection](https://www.mediawiki.org/wiki/Extension:PlaceNewSection) to point to this repo, which I merely referred to on the Talk: page. So this comes as a bit of a surprise to me.

Is there a go-to reference that explains what the current "MW standards" are, or could you please summarize what needed to be changed?

**Comment by ernstki** [comment]
> Is there a go-to reference that explains what the current "MW standards" are, or could you please summarize what needed to be changed?

…apart from the obvious, like whitespace and PHPDoc directives, or whatever those are.

**Comment by reedy** [comment]
It's from the request at https://phabricator.wikimedia.org/T344501

Which results in https://www.mediawiki.org/wiki/Writing_an_extension_for_deployment#Preparing_for_deployment needing to be followed.

Which in turn, this PR is basically doing various things from https://www.mediawiki.org/wiki/Best_practices_for_extensions which would be needed eventually if deployment was going to go ahead

**Comment by ernstki** [comment]
OK, terrific. Thanks for your explanation and your patience.

Will the end goal be to get this in MediaWiki's Gerrit, in the official GitHub org, and ExtensionDistributor, or are those parallel goals?

**Comment by reedy** [comment]
> Will the end goal be to get this in MediaWiki's Gerrit

In theory, if it moves forward for deployment

>in the official GitHub org, and ExtensionDistributor, or are those parallel goals?

If it ends up in Gerrit, both of those basically happen as side effects

File: i18n/en.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+	"placenewsection-desc": "Adds the <nowiki>__ADDNEWSECTIONBELOW__</nowiki> and <nowiki>__ADDNEWSECTIONABOVE__</nowiki> magic words to control new section placement\""
+}
```
Comment (ernstki) [inline i18n/en.json:3]: @reedy Are missing the missing LFs at EOF in these files any cause for concern, from your perspective?

I assume it's benign in this context, but I can think of some situations (in a Unix environment) where it's not.

**Comment by ernstki** [comment]
@reedy It wasn't so scary once I told GitHub to ignore whitespace changes. Thanks for getting us back on track with current standards. I'll make sure I'm tuned in to the Phabricator issue(s), so I can watch how it progresses from here.

**Pull request merged**